cmp/grid/renderers/MultiFieldRenderer.ts

```diff
@@ -30,25 +30,37 @@ export function multiFieldRenderer(value, context): ReactNode {
     );
 
     const {mainRenderer, delimiter, subFields = []} = multiFieldConfig,
-        [topFields, bottomFields] = partition(subFields, it => it.position === 'top'),
-        topRowItems = [],
+        [topFields, bottomFields] = partition(subFields, it => it.position === 'top');
+
+    let topRowItems = [],
         bottomRowItems = [];
 
     // Render main field to top row
     topRowItems.push(renderMainField(value, mainRenderer, context));
 
     // Render SubFields to top row
-    topFields.forEach(it => {
-        if (delimiter) topRowItems.push(renderDelimiter(delimiter));
+    topFields.forEach((it, idx) => {
         topRowItems.push(renderSubField(it, context));
     });
 
+    pull(topRowItems, null);
+
     // Render SubFields to bottom row
     bottomFields.forEach((it, idx) => {
-        if (delimiter && idx > 0) bottomRowItems.push(renderDelimiter(delimiter));
         bottomRowItems.push(renderSubField(it, context));
     });
 
+    pull(bottomRowItems, null);
+
+    // Insert delimiter if applicable
+    if (delimiter) {
+        const insertDelimiterFn = (value, idx, array) => {
```

Comment:
we have a new util called "intersperse" for this

Reply:
cool